test/unit/react-imgix.test.jsx

```diff
@@ -309,6 +309,75 @@ describe("When in <source> mode", () => {
     const onMountArg = onMountedSpy.lastCall.args[0];
     expect(onMountArg).toBeInstanceOf(HTMLSourceElement);
   });
+
+  describe("using the htmlAttributes prop", () => {
+    it("assigns an alt attribute given htmlAttributes.alt", async () => {
+      const htmlAttributes = {
+        alt: "Example alt attribute"
+      };
+
+      sut = mount(
+        <Source
+          src={"https://mysource.imgix.net/demo.png"}
+          sizes="100vw"
+          htmlAttributes={htmlAttributes}
+        />
+      );
+
+      expect(sut.props()["htmlAttributes"].alt).toEqual(htmlAttributes.alt);
+    });
+
+    it("passes any attributes via htmlAttributes to the rendered element", () => {
+      const htmlAttributes = {
+        "data-src": "https://mysource.imgix.net/demo.png"
+      };
+      sut = mount(
+        <Source
+          src={"https://mysource.imgix.net/demo.png"}
+          sizes="100vw"
+          htmlAttributes={htmlAttributes}
+        />
+      );
+
+      expect(sut.props()["htmlAttributes"]["data-src"]).toEqual(
+        htmlAttributes["data-src"]
+      );
+    });
+
+    it("attaches a ref via htmlAttributes", () => {
+      const callback = sinon.spy();
+
+      sut = mount(
+        <Source
+          src={"https://mysource.imgix.net/demo.png"}
+          sizes="100vw"
+          htmlAttributes={{ ref: callback }}
+        />
+      );
+
+      expect(callback.called).toBeTruthy();
+      expect(callback.callCount).toEqual(1);
+    });
+
+    it("stills calls onMounted if a ref is passed via htmlAttributes", () => {
+      const htmlAttrCallback = sinon.spy();
+      const onMountedCallback = sinon.spy();
+
+      sut = mount(
+        <Source
+          src={"https://mysource.imgix.net/demo.png"}
+          sizes="100vw"
+          onMounted={onMountedCallback}
+          htmlAttributes={{ ref: htmlAttrCallback }}
+        />
+      );
+
+      expect(htmlAttrCallback.called).toBeTruthy();
+      expect(htmlAttrCallback.callCount).toEqual(1);
+      expect(onMountedCallback.called).toBeTruthy();
+      expect(onMountedCallback.callCount).toEqual(1);
```

Comment:
nit: `called` and `callCount` are overlapping assertions - we can just drop the `.called` assertion as the `.callCount` tests for the same thing.

Reply:
Ah forgot to take those out. Thanks